python/lsst/ts/mtdometrajectory/dome_trajectory.py

```diff
@@ -478,18 +485,25 @@ async def follow_target(self):
         if (
             self.move_dome_elevation_task.done()
             and self.dome_remote.evt_elMotion.has_data
+            and self.enable_el_motion is True
         ):
             dome_target_elevation = self.get_dome_target_elevation()
             desired_dome_elevation = self.algorithm.desired_dome_elevation(
                 dome_target_elevation=dome_target_elevation,
                 telescope_target=self.telescope_target,
                 next_telescope_target=self.next_telescope_target,
             )
-            if desired_dome_elevation is not None:
+            if desired_dome_elevation is not None and not math.isnan(
```

Comment:
Consider using `math.isfinite()` instead of `not math.isnan()`. It will cover more cases and also avoids the double comparison.